src/domain-services/flows/flow-service.ts

```diff
@@ -322,28 +323,39 @@ export class FlowService {
     const entityCondKey = orderBy.entity as unknown;
     const entityCondKeyFlowObjectType = entityCondKey as FlowObjectType;
 
-    flowObjects = await database.flowObject.find({
-      where: {
-        objectType: entityCondKeyFlowObjectType,
-        refDirection,
-        objectID: {
-          [Op.IN]: entityIDsSorted,
-        },
-      },
-      distinct: ['flowID', 'versionID'],
-    });
+    // Order map
+    const orderMap = new Map<number, number>();
+    for (const [index, entityID] of entityIDsSorted.entries()) {
+      orderMap.set(entityID, index);
+    }
 
+    const chunks = splitIntoChunks(entityIDsSorted, 1000);
+    // Instead of doing a single query that may end up on a 'Memory Error'
+    // we will do a progressive search
+    // by chunks of 1000
+    for (const chunk of chunks) {
+      const flowObjectsBatch = await database.flowObject.find({
+        where: {
+          objectType: entityCondKeyFlowObjectType,
+          refDirection,
+          objectID: {
+            [Op.IN]: chunk,
+          },
+        },
+        distinct: ['flowID', 'versionID'],
+      });
+      flowObjects.push(...flowObjectsBatch);
+    }
```

Comment:
We usually parallelize the fetching with `Promise.all()`. By doing this, you can immediately assign the results into a variable, so that you don't have to define `flowObjects` above as you do, since it is typed as `any[]` if you do so, which is something we don't want.

```suggestion
    const flowObjects = (
      await Promise.all(
        splitIntoChunks(entityIDsSorted, PG_MAX_QUERY_PARAMS - 2).map(
          (entityIds) =>
            database.flowObject.find({
              where: {
                objectType: entityCondKeyFlowObjectType,
                refDirection,
                objectID: {
                  [Op.IN]: entityIds,
                },
              },
              distinct: ['flowID', 'versionID'],
            })
        )
      )
    ).flat();
```